### ISSUE TITLE
Endless updating channel on Safari 11.0.3

### ISSUE THREAD

**Issue by mrpau-richard** [learningequality/kolibri#5562]
### Observed behavior
After I updated and published the title of my own channel at the Studio, and update the old version in my local it will have an endless updating channel for just a title update I waited more than 10 minutes.

When I manually refresh the page the channel title and version are updated in my local.

I tried this on Firefox browser and the updating channel is working correctly this issue is only on Safari 11.0.3.

### Expected behavior
Updating channel should work in the Safari 11.0.3

### Errors and logs
![Screen Shot 2019-05-28 at 2 00 04 PM](https://user-images.githubusercontent.com/8663934/58455484-66983d80-8154-11e9-9ac8-b57d7e4eb6f5.png)

Please wrap errors in triple backticks for clean formatting like this:
```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (user_agent-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (coach_side_nav-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (user_module_user_profile_nav_side_nav-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (facility_management_side_nav-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (user_module_login_nav_side_nav-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (device_management_side_nav-0.12.4b4.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (learn_module_side_nav-0.12.4b4.js.map, line 0)
```

### Steps to reproduce
* Update your own channel tittle at the Studio.
* Assuming that you already imported the old version of this channel in your local
* Update the channel with the new version
* The endless updating channel show up.

### Context

Tell us about your environment, including:
 * Kolibri version: 0.12.4b4
 * Operating system: iOS
 * Browser:  Safari 11.0.3

**Comment by jonboiser** [comment]
I'm able to reproduce this as well. Tried it on Chrome ✅ 	and Safari ❌ .

On Safari, the issue seems to be that the UI is not correctly reacting (by closing the progress bar) when it notices the the `REMOTECHANNELIMPORT` Task reaches the `COMPLETED` status.

This might be another issue with getting the timing/ordering right with these Tasks. And specifically with the way Promises are working on Safari.

**Comment by bjester** [comment]
On latest pull of `release-0.12.y` branch, I can no longer reproduce this exactly, but there's a new issue. After running the update, the update isn't reflected on the front-end; I still see the old version of the channel. After refreshing the page, I see the updated channel. In the intermediary state where it's updated but not showing the updated content, I can click "Update" again and it will get into the stuck endless update loop, presumably because there was nothing to update. This only happens in Safari for me.

**Comment by bjester** [comment]
From my investigation, the reload of the page route isn't triggering the `SET_CHANNEL_LIST` on the state in Safari and therefore it is reusing the old state.

**Comment by rtibbles** [comment]
Hrm - any ideas why Safari is doing this? Are we missing a polyfill that it needs?

**Comment by bjester** [comment]
It appears that state isn't reset which probably relates to how Safari is handling the route change specifically. [This line](https://github.com/learningequality/kolibri/blob/release-v0.12.y/kolibri/plugins/device_management/assets/src/routes/wizardTransitionRoutes.js#L34) is stopping it from reloading the page, but even if that was bypassed, also within `showSelectContentPage`, [here](https://github.com/learningequality/kolibri/blob/release-v0.12.y/kolibri/plugins/device_management/assets/src/modules/wizard/handlers.js#L179) it wouldn't set the channel. I'm curious why it doesn't set the channel there because it doesn't do anything else, so it would seem that it could bypass the request entirely instead of just the response.

**Comment by bjester** [comment]
Would it be terrible to do `window.location.reload()` instead of `this.$router.go()`?

**Comment by rtibbles** [comment]
The main difference is that it would do a full page refresh (rerequest the page from the server), rather than just re-trigger the navigation route.

However, as we are caching the server pages now (as per #5613 and previous work), it should be relatively painless.

**Comment by bjester** [comment]
Yep, it would do the full page refresh, but the user experience shouldn't be terribly reduced.

For additional context, I forgot I had made a change which I now believe fixed the original issue, which was to not send any arguments go `$router.go()` since it's supposed to take a numerical argument that specifies how many steps in history to go forward or backward. It was sending the current route object previously.

**Comment by rtibbles** [comment]
Would it work to just do a router navigate to the current page, rather than traversing page history?

**Comment by bjester** [comment]
@rtibbles You mean something like `$router.replace($router.currentRoute)`?

**Comment by rtibbles** [comment]
Yeah

**Comment by bjester** [comment]
Yeah I had tried that. That actually reproduces the original issue in all browsers. I think more work to manually "refresh" the page would be required to get that to work as expected

**Comment by rtibbles** [comment]
Similarly if you do `router.push`?

**Comment by bjester** [comment]
Yep

**Comment by rtibbles** [comment]
Fixed in #5615
